packages/resolvers/glob/src/GlobResolver.js

```diff
@@ -102,14 +102,30 @@ export default (new Resolver({
 
       if (!result || !result.filePath) {
         throw errorToThrowableDiagnostic(
-          `Unable to resolve ${pkg} from ${sourceFile} when evaluating specifier ${specifier}`,
+          `Unable to resolve ${pkg} from ${sourceFile} when resolving specifier ${specifier}`,
           dependency,
         );
       } else if (result.diagnostics) {
         throw new ThrowableDiagnostic({diagnostic: result.diagnostics});
       }
+      // We keep Flow happy by copying the filePath at a point we know it's defined
+      const filePath = result.filePath;
 
-      specifier = path.resolve(path.dirname(result.filePath), rest);
+      const ctx = {
```

Comment:
I wasn't sure what to do about this - `findPackage` requires a context, so I provided one with the invalidation info from the step above, however when we return the resolution at the end of this method around line 178, there's only `invalidateOnFileCreate: [{glob: normalized}],` included. Should we collect all these additional invalidations and add them there as well?